### PR TITLE
Introducing the SunSkyStage

### PR DESCRIPTION
**Pull request by samcake**
- Adding the SunSkyStage object in the library "model", it provides and update the graphics objects required to render a pseudo realistic sun/sky model. The main sun light object is updated based on the "origin position" on earth and the current time of the day and year. The sun direction is considered constant all over the domain because of the current scale of objects.

- Exposed the SUnSKyStage api through a simplified set of calls on a new scripting entry point called "Scene"

- added an example script to show how one can drive the sun/sky settings in a server from a script

### CODE REVIEW

**Comment by samcake** [comment]
OK hold on a sec, i fucked up again...

**Comment by samcake** [comment]
good to go

**Comment by hifi-gustavo** [comment]
Build console output for commit: fcb293d0b2155d4334610dc6448a2073356bbca7 available at: https://gist.github.com/293c90c2e2e6a3e8aa3d<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by nbq** [comment]
@samcake the links are not pointing to a valid file. Can you please request Jenkins to build again? Thanks!

**Comment by hifi-gustavo** [comment]
Build console output for commit: b04dbf7a34b85ee5c013e12aedd0f66f1f14dbee available at: https://gist.github.com/6a8e92807014b0f21cb9<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by nbq** [comment]
Nevermind, took a bit to come through, got it @samcake

File: examples/example/misc/sunLightExample.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+var longitude = 0.0;
+var latitude = -45.0;
+
+Scene.setDayTime(hour);
+Scene.setOriginLocation(longitude, latitude, 0.0);
+
+function ticktack() {
+	hour += 0.1;
+	//Scene.setSunIntensity(Math.cos(time));
+	if (hour > 24.0) {
```
Comment (ZappoMan) [inline examples/example/misc/sunLightExample.js]: indent looks broken

**Comment by samcake** [comment]
Brad,

Also i m very open to debate about the name i picked
IN particular the "SceneScriptingInterface" i don;t like this name but just picked this to move fast
CHeers,

S

**Comment by nbq** [comment]
Since the last commits were just tab fixes far as I can tell, here is my results: 

**OSX Test**

- Script Tested: https://raw.githubusercontent.com/samcake/hifi/b4cb5ca9d1e455ec7425a950259bdbc7bb65c9f7/examples/example/misc/sunLightExample.js
- Script Result: http://gyazo.com/87325559fe940969ffbf1bc62b27ac3f

*OSX QA Response: Works as expected*

Will test next build on Windows.

**Comment by hifi-gustavo** [comment]
Build console output for commit: bc8d688cdea8dab4871523a66c6a123198b6ecc2 available at: https://gist.github.com/8e50b1f4351c5b96a6a9<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 96d5a2ab4221e214d364e4437374a5059454209f available at: https://gist.github.com/48c4b45e1c4876cec56c<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by OmegaHeron** [comment]
Really wanted to test/experiment with this - but windows builds show missing keys - outputting xml file unlike (useless to me) os x builds which offer binary.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e3b3c8e1ef8f23324f188efd7f01ad5f9e3b34f5 available at: https://gist.github.com/0628fd3211adbbea03de<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e3b3c8e1ef8f23324f188efd7f01ad5f9e3b34f5 available at: https://gist.github.com/2bcd56f3829b15d36ea0<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by samcake** [comment]
*It works thanks!*

**Comment by samcake** [comment]
PLease hold on for merge, still refining

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e3b3c8e1ef8f23324f188efd7f01ad5f9e3b34f5 available at: https://gist.github.com/b62ddd6f45257024d51b<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f6a9bd187090049e807c8ff51e32d3d5647ea276 available at: https://gist.github.com/b539926a96be59ca2e4a<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 74f0d98fd6ff65ae26dfda6374edc03c2a1d5ec9 available at: https://gist.github.com/482198f7a6ed28607635<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c7eeebe52cc4b030a6bd6f3436497b4ee8047426 available at: https://gist.github.com/e3303decacb5708064c6<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4347.dmg

**Comment by OmegaHeron** [comment]
It's looking great @samcake - looking forward to being able to tie my script that is waiting and ready with my RL geo location and time ;)  Once it's done - will the script be able to run on server side offering all visitors the same presentation of sun position?

**Comment by samcake** [comment]
Thanks Heron!

Yep we ll try to make a function that get s callled when once login in a server coming from the server
Like the chess game in "appartment"

Glad you like

**Comment by samcake** [comment]
Ready for review and merge, thanks

**Pull request merged**